src/stackwalk.c

```diff
@@ -88,6 +88,9 @@ static int jl_unw_stepn(bt_cursor_t *cursor, jl_bt_element_t *bt_data, size_t *b
     jl_set_safe_restore(&buf);
     if (!jl_setjmp(buf, 0)) {
 #endif
+        jl_bt_element_t *threadid_entry = bt_data + n;
+        threadid_entry->uintptr = jl_threadid() + 1; // add 1 as 0 is preserved to indicate start of backtrace
```

Comment:
I am not sure we want to do it here, since this is also used outside the profiler.
I was thinking directly in `signals-unix.c`